### PR TITLE
Align carbon_ch with Python implementation

### PR DESCRIPTION
**Pull request by github-vincent-miszczak**
`carbon-relay-ng` has incorrect `carbon_ch` implementation.

It lacks https://github.com/graphite-project/carbon/commit/024f9e67ca47619438951c59154c0dec0b0518c7#diff-1486787206e06af358b8d935577e76f5 update

It fixes https://github.com/graphite-ng/carbon-relay-ng/issues/335

### CODE REVIEW

**Comment by Dieterbe** [comment]
for the record, the problem and fix in python are described at https://github.com/graphite-project/carbon/pull/196

File: route/consistent_hashing.go [hunk 1]
```diff
@@ -85,6 +86,18 @@ func (h *ConsistentHasher) AddDestination(d *dest.Destination) {
 		keyBuf.WriteString(":")
 		keyBuf.WriteString(strconv.Itoa(i))
 		position := computeRingPosition(keyBuf.Bytes())
+		//Align with https://github.com/graphite-project/carbon/commit/024f9e67ca47619438951c59154c0dec0b0518c7#diff-1486787206e06af358b8d935577e76f5
+	outer:
```
Comment (robert-milan) [inline route/consistent_hashing.go]:
```suggestion
	OUTER:
```

File: route/consistent_hashing.go [hunk 2]
```diff
@@ -85,6 +86,18 @@ func (h *ConsistentHasher) AddDestination(d *dest.Destination) {
...
+		for {
+			for i := 0; i < len(h.Ring); i++ {
+				if position == h.Ring[i].Position {
+					position++
+					continue outer
```
Comment (robert-milan) [inline route/consistent_hashing.go]:
```suggestion
					continue OUTER
```

File: route/consistent_hashing.go (hunk 1 above)
Comment (Dieterbe) [reply route/consistent_hashing.go]: seems like the go best practice is camelcased label names
https://golang.org/doc/effective_go.html#switch

**Comment by loitho** [comment]
Hi there, 
is it possible to merge this fix ? The issue with the hashing prevents us from using the Carbonate tool which is annoying as it's very useful.

Kind regards

**Comment by robert-milan** [comment]
I don't currently have time to write the unit tests needed before merging this.

**Comment by Dieterbe** [comment]
we shouldn't break carbon-relay-ng deployments when they upgrade.
so I think we should have a config flag to control the sharding logic (by default: backwards compat with carbon-relay-ng's previous behavior, and opt-in for the new "compatible with python relay" behavior), at some point we can do the breaking change of flipping the default when we do a major release. (for extra benefit, at that point, people will have had a chance to test the new behavior)

**Comment by Dieterbe** [comment]
#477 is the reworked version of this PR.

**Comment by Dieterbe** [comment]
See #477 - thanks for the effort on this!

**Pull request closed without merging**